File: 165. Compare Version Numbers/Solution.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        v1 = version1.split('.')
        v2 = version2.split('.')

        v = min(len(v1), len(v2))
        for i in range(v):
            if int(v1[i]) > int(v2[i]):
                return 1
            elif int(v2[i]) > int(v1[i]):
                return -1
        if len(v1) == len(v2):
            return 0
        else:
            if len(v1)>len(v2):
                a = ''.join(v1[len(v2):])
                if int(a) > 0:
                    return 1
                else:
                    return 0
            else:
                a = ''.join(v2[len(v1):])
                if int(a) > 0:
                    return -1
                else:
                    return 0
                

File: 165. Compare Version Numbers/Solution.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {
        int i = 0, j = 0;
        int n = version1.size(), m = version2.size();
        
        while (i < n || j < m) {
            int num1 = 0, num2 = 0;
            
            // take the number until '.' in version1
            while (i < n && version1[i] != '.') {
                num1 = num1 * 10 + (version1[i] - '0');
                i++;
            }
            
            // take the number until '.' in version2
            while (j < m && version2[j] != '.') {
                num2 = num2 * 10 + (version2[j] - '0');
                j++;
            }
            
            // compare this revision
            if (num1 < num2) return -1;
            if (num1 > num2) return 1;
            
            // skip the dot
            i++;
            j++;
        }
        
        return 0;
    }
};
